sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamReader.scala

```diff
@@ -173,15 +173,13 @@ final class DataStreamReader private[sql](sparkSession: SparkSession) extends Lo
       case _ => None
     }
     ds match {
-      case provider: TableProvider =>
+      // file source v2 does not support streaming yet.
```

Comment:
file source v2 is never supported in streaming, as `FileTable.capabilities` doesn't include streaming ones.

The reason I check it earlier is: now we call `TableProvider.inferSchema` before checking table capabilities, and the error becomes different if `path` is not specified.

It's weird that file source reports different error between batch and streaming when path is not specified. We should unify it. Here I just don't want to be blocked by an existing problem.  cc @gengliangwang